### PR TITLE
libdb: fix AppleClang >=12 + cross-build to iOS + bump msys2 + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

closes https://github.com/conan-io/conan-center-index/issues/7917

we may have to raise if Visual Studio 2019 until a fix is found.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@danimtb @jgsogo @SSE4 @uilianries 

Another expired Visual Studio license

```
Copyright (C) Microsoft Corp. All rights reserved.

The license for Visual Studio has expired.

The evaluation period for this product has ended.
```

File: recipes/libdb/all/conanfile.py [hunk 1]
```diff
@@ -47,33 +62,49 @@ def config_options(self):
...
+        if not self.options.get_safe("with_cxx", False):
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_tcl:
+            self.requires("tcl/8.6.10")
+
+    def validate(self):
         if self.options.get_safe("with_cxx"):
```
Comment (SSE4) [inline recipes/libdb/all/conanfile.py:74]:
```suggestion
        if self.settings.compiler == 'Visual Studio':
            raise ConanInvalidConfiguration('This compiler is unsupported')
        if self.options.get_safe("with_cxx"):
```

Comment (SpaceIm) [reply recipes/libdb/all/conanfile.py:74]: It is supported.

**Review by SSE4 — APPROVED** [review]

File: recipes/libdb/all/conanfile.py [hunk 2]
```diff
@@ -47,33 +62,54 @@ def config_options(self):
...
+
+    def requirements(self):
+        if self.options.with_tcl:
+            self.requires("tcl/8.6.10")
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            # FIXME: it used to work with previous versions of Visual Studio 2019 in CI of CCI.
+            if tools.Version(self.settings.compiler.version) == "16":
+                raise ConanInvalidConfiguration("Visual Studio 2019 not supported.")
```
Comment (prince-chrismc) [inline recipes/libdb/all/conanfile.py:77]:
```suggestion
                raise ConanInvalidConfiguration("Visual Studio 2019 is currently not supported by this recipe.")
```

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**